base/reduce.jl

```diff
@@ -870,12 +870,15 @@ end
 """
     findmax(f, domain) -> (f(x), index)
 
-Returns a pair of a value in the codomain (outputs of `f`) and the index of
+Returns a pair of a value in the codomain (outputs of `f`) and the index or key of
```

Comment:
Indicies are a special case of keys. For Arrays they are the same, but Dicts have keys but not indices. If we just mentioned keys here, that would be correct but obtuse because folks shouldn't need to know that Arrays as have keys to understand `findmax(::Function, ::Array)`.

I'm not sure if the difference is explicitly mentioned or explained elsewhere, but I don't think this PR is the place for it.